environment/src/main/java/com/sequenceiq/environment/environment/encryption/EnvironmentEncryptionService.java

```diff
@@ -72,17 +68,19 @@ EncryptionResources getEncryptionResources(String cloudPlatform) {
 
     @VisibleForTesting
     DiskEncryptionSetCreationRequest createEncryptionResourcesCreationRequest(EnvironmentDto environment) {
+        String encryptionKeyResourceGroupName = environment.getParameters().getAzureParametersDto().getAzureResourceEncryptionParametersDto()
+                .getEncryptionKeyResourceGroupName();
         DiskEncryptionSetCreationRequest.Builder builder = new DiskEncryptionSetCreationRequest.Builder()
                 .withId(Crn.safeFromString(environment.getResourceCrn()).getResource())
                 .withCloudCredential(credentialToCloudCredentialConverter.convert(environment.getCredential()))
                 .withTags(environmentTagProvider.getTags(environment, environment.getResourceCrn()))
                 .withCloudContext(getCloudContext(environment))
-                .withEncryptionKeyUrl(environment.getParameters().getAzureParametersDto().getAzureResourceEncryptionParametersDto().getEncryptionKeyUrl());
-        if (isSingleResourceGroup(environment)) {
-            builder.withSingleResourceGroup(true);
-            builder.withResourceGroupName(environment.getParameters().getAzureParametersDto().getAzureResourceGroupDto().getName());
+                .withEncryptionKeyUrl(environment.getParameters().getAzureParametersDto().getAzureResourceEncryptionParametersDto().getEncryptionKeyUrl())
+                .withDiskEncryptionSetResourceGroupName(environment.getParameters().getAzureParametersDto().getAzureResourceGroupDto().getName());
```

Comment:
Extract `environment.getParameters().getAzureParametersDto().getAzureResourceGroupDto().getName()` into a variable since it is also used in line 83.

Also be aware that `getAzureResourceGroupDto()` may return `null` when using Multi-RG. As I see, we are not enforcing Single-RG for DES in `AzureParameterValidator` (environment), thanks to the changes in #10756. (I know I asked for rearranging the `validateEncryptionParameters()` call. 😄 ) Handling Multi-RG correctly would need further work, so I suggest adding an explicit check there that fails when "DES is given and Multi-RG". Please verify this bad behavior first before making the change just to be certain.

Reply:
I see.
In order to fix this for multi-RG, raised CB-13130.
Thank you for highlighting this issue +1